app/lib/report_helper.rb

```diff
@@ -865,20 +857,28 @@ def apply_filters!(rows, tax_2d, all_genera, params)
   def taxonomy_details(pipeline_run_id, background_id, params)
     # Fetch and clean data.
     t0 = wall_clock_ms
-    tax_2d = cleanup_all!(convert_2d(fetch_taxon_counts(pipeline_run_id, background_id)))
+    taxon_counts = fetch_taxon_counts(pipeline_run_id, background_id)
+    tax_2d = taxon_counts_cleanup(taxon_counts)
     t1 = wall_clock_ms
 
     # These counts are shown in the UI on each genus line.
     count_species_per_genus!(tax_2d)
 
-    # Pull out all genera names in sample (before filters are applied).
-    all_genera = Set.new
-    tax_2d.each do |_tax_id, tax_info|
-      tax_name = tax_info['name']
-      tax_level = tax_info['tax_level']
-      all_genera.add(tax_name) if tax_level == TaxonCount::TAX_LEVEL_GENUS
+    # Generate lineage info.
+    unfiltered_ids = []
+    tax_2d.each do |tid, _|
+      unfiltered_ids << tid
+    end
+    lineages = TaxonCount.connection.select_all(TaxonLineage.where(taxid: unfiltered_ids)).to_hash
```

Comment:
similar to my commebt above. This Taxon.connection.selecg_all is not necessary. to_hash is also not.

Reply:
The lineage info needs to come from TaxonCounts because "-200 and -100 are usually from the lineage table not from taxon counts. In taxon counts they will be like -200000543 and -10000543."  cc: Boris